### PR TITLE
[ML-151] enable Summarizer with OAP

### PR DESCRIPTION
**Pull request by minmingzhu**
1. enable Summarizer CPU and GPU with one API
2. add Summarizer example

### CODE REVIEW

File: mllib-dal/src/spark-3.1.1/main/scala/org/apache/spark/mllib/stat/Statistics.scala [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  : KolmogorovSmirnovTestResult = {
+    KolmogorovSmirnovTest.testOneSample(data, distName, params: _*)
+  }
+
+  /**
+    * Java-friendly version of `kolmogorovSmirnovTest()`
+    */
+  @Since("1.5.0")
+  @varargs
+  def kolmogorovSmirnovTest(
```
Comment (xwu99) [inline mllib-dal/src/spark-3.1.1/main/scala/org/apache/spark/mllib/stat/Statistics.scala:269]:
```suggestion
  def kolmogorovSmirnovTest(data: JavaDoubleRDD,
```
should format like this

**Review by xwu99 — CHANGES_REQUESTED** [review]
Pls rebase to master as there has been some large refactor.

File: mllib-dal/src/main/scala/com/intel/oap/mllib/stat/SummarizerDALImpl.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      val computeEndTime = System.nanoTime()
+
+      val durationCompute = (computeEndTime - computeStartTime).toDouble / 1E9
+
+      logInfo(s"SummarizerDAL compute took ${durationCompute} secs")
+
+      val ret = if (OneCCL.isRoot()) {
+
+        val convResultStartTime = System.nanoTime()
+        val meanMatrix = OneDAL.numericTableToOldMatrix(OneDAL.makeNumericTable(result.meanNumericTable))
```
Comment (xwu99) [inline mllib-dal/src/main/scala/com/intel/oap/mllib/stat/SummarizerDALImpl.scala:75]: The code is not decent here to convert twice. 
There is a logical problem in the function numericTableToOldMatrix. isTransposed should be set to true. This function is not used, you can remove it. 
Is result.meanNumericTable a Nx1 or 1xN matrix? For Nx1, You can call numericTableNx1ToVector, For 1xN, you can modify numericTableNx1ToVector function to numericTable1xNToVector.

File: mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark320/Statistics.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import org.apache.spark.storage.StorageLevel
+
+import scala.annotation.varargs
+
+
+/**
+  * API for statistical functions in MLlib.
+  */
+@Since("1.1.0")
+class Statistics extends SummarizerShim{
```
Comment (xwu99) [inline mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark320/Statistics.scala:41]:
```suggestion
class Statistics extends SummarizerShim {
```

File: mllib-dal/src/main/scala/com/intel/oap/mllib/stat/SummarizerShim.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+package com.intel.oap.mllib.stat
+
+import org.apache.spark.{SPARK_VERSION, SparkException}
+import org.apache.spark.internal.Logging
+import org.apache.spark.mllib.linalg.Vector
+import org.apache.spark.mllib.stat.MultivariateStatisticalSummary
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Dataset}
+
+import org.apache.spark.mllib.stat.spark320.{Statistics => SummarizerSpark320 }
```
Comment (xwu99) [inline mllib-dal/src/main/scala/com/intel/oap/mllib/stat/SummarizerShim.scala:26]: better to use Statistics => StatisticsSpark320

**Review by xwu99 — CHANGES_REQUESTED** [review]

File: mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark320/Statistics.scala (hunk 3 above)
Comment (xwu99) [reply mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark320/Statistics.scala:41]: something wrong with CI, pls make sure you didn't mess up other algos.  Only change what you need to change. Pls compare the changed files for 3.1.1, 3.1.2 and 3.2 to see anything inconsistent.

File: dev/codestyle/.clang-format [hunk 5]
```diff
@@ -2,29 +2,24 @@
 Language:        Cpp
```
[the remark this answers is not in the document]
Comment (xwu99) [reply dev/codestyle/.clang-format:2]: for code format change, you just need to format your changes.

[the remark this answers is not in the document]
Comment (xwu99) [reply dev/codestyle/.clang-format:2]: for code format change, you just need to format your changes.

**Review by xwu99 — CHANGES_REQUESTED** [review]

**Comment by minmingzhu** [comment]
> Thanks for opening a pull request!
> 
> Could you open an issue for this pull request on Github Issues? https://github.com/oap-project/oap-mllib/issues
> 
> Then could you also rename _**pull request title**_ and _**commit log**_ in the following format?
> 
> ```
> [ML-${ISSUES_ID}] ${detailed message}
> ```
> 
> See also:
> 
> * [Other pull requests](https://github.com/oap-project/oap-mllib/pulls/)

**Comment by minmingzhu** [comment]
#151

**Comment by minmingzhu** [comment]
#153

**Comment by xwu99** [comment]
@minmingzhu thanks for your work. Merged!

**Pull request merged**